### PR TITLE
Flag to disable DNS record creation

### PR DESCRIPTION
**Pull request by samholmes**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201256727099464

### CODE REVIEW

File: provisions/digitalocean/util.ts [hunk 1]
```diff
@@ -28,13 +29,15 @@ export interface ProvisionSettings {
...
   DOMAIN_INTERNAL: string;
+  SKIP_DNS: boolean;
   VOLUME_SIZE: number;
   REGION: string;
   SIZE: string;
   SSH_KEY_IDS: number[];
 }
 
 export async function getProvisionSettings(
+  config: Config,
```
Comment (samholmes) [inline provisions/digitalocean/util.ts:40]: So the motivation here is to deprecate `ProvisionOptions` in favor of just accepting a `Config` parameter (less type transformations; simpler codebase this way). Perhaps a deprecated comment with a "TODO" is warranted. Or, we can add the field to the existing type for now and then save this effort/motivation for later. I'm open to suggestions.

File: provisions/digitalocean/index.ts [hunk 2]
```diff
@@ -72,11 +72,10 @@ if (config?.installScripts != null) {
 
       // Env Var:
       const { TLD } = settings;
-      const COUCH_MODE =
-        config?.couchClusterSeedList.length != null &&
+      const COUCH_MODE = config?.couchClusterSeedList.length != null &&
```
Comment (swansontec) [inline provisions/digitalocean/index.ts:75]: Now that `config` & `couchClusterSeedList` are always present, we need to change this to `length > 0` and drop the null checks.

**Review by swansontec — CHANGES_REQUESTED** [review]
Just noticed one little thing to fix.

File: provisions/digitalocean/index.ts (hunk 2 above)
Comment (samholmes) [reply provisions/digitalocean/index.ts:75]: Good catch

Comment (samholmes) [reply provisions/digitalocean/index.ts]: I fixed up d701b62 to include the null-check removal for `config`.

**Review by swansontec — APPROVED** [review]

**Pull request merged**